src/pages/nft/gift/index.vue

```diff
@@ -91,6 +104,12 @@ export default {
     classId() {
       return this.$route.query.class_id;
     },
+    collectionId() {
+      return this.$route.query.collection_id;
+    },
+    primaryKey() {
+      return this.collectionId || this.classId;
+    },
```

Comment:
This should put in `nftOrCollectionMixin`?

Reply:
yes in fact it already exists in there...